kivy/core/window/_window_sdl2.pyx

```diff
@@ -686,3 +707,36 @@ cdef SDL_Surface* flipVert(SDL_Surface* sfc):
         memcpy(&rpixels[pos], &pixels[(pxlength - pos) - pitch], pitch)
 
     return result
+
+cdef SDL_Surface* surface_from_core_image(core_image):
+    # Create an SDL_Surface from a core image's texture pixels. Use
+    # SDL_FreeSurface(surface) to deallocate.
+    #
+    # Returns NULL if the core image has no texture pixels, or if
+    # there's a problem with the creation of an SDL Surface.
+    cdef int width, height
+    width, height = core_image.size
+    cdef int pitch = width * 4
+    cdef bytes pixels
+    cdef char* sdl_error
+    try:
+        pixels = core_image.image.texture.pixels
+    except AttributeError:
+        return NULL
+
+    if pixels == None:
```

Comment:
Maybe use `if not pixels:` instead of testing for None explicitly? Just a safety measure, in case for whatever reason you get an empty byte array or whatever from the coreimage.

Reply:
My 2¢ to be honest I prefer explicit rather than implicit, I rather not covering the case and let it crash to later (unit test and) fix it, than covering some cases "just in case" even though they may never happens. And this exactly for the reason you evoked "whatever reason you get an empty byte array or whatever". I think if we are not sure of what we get and if we don't make it explicit we'll pay for it later. But I believe it's also a matter of taste and point of view..
Same thing with catching `Exception`, I'm not so much of a big fan.
Other matter of taste: I would use a more Pythonic way `if pixels is None`.